crux_kv/src/lib.rs

```diff
@@ -105,17 +105,14 @@ where
 
     /// Read a value under `key`, will dispatch the event with a
     /// `KeyValueResult::Get { value: Vec<u8> }` as payload
-    pub fn get<F>(&self, key: String, make_event: F)
+    pub fn get<F>(self, key: String, make_event: F)
     where
         F: FnOnce(Result<Option<Vec<u8>>, KeyValueError>) -> Ev + Send + Sync + 'static,
     {
-        self.context.spawn({
-            let context = self.context.clone();
-            let this = self.clone();
-
+        self.context.clone().spawn({
             async move {
-                let response = this.get_async(key).await;
-                context.update_app(make_event(response));
```

Comment:
Given that `get_async` basically only needs the `context` would it not be nicer to just avoid cloning `self` entirely?

Could extract a standalone `fn get_async(context: CapabilityContext<KeyValueOperation, Ev>), call that with a cloned context and keep the `&self`?

Reply:
Oh yeah that'd be loads better. I will do, Thanks 🤩

Reply:
much nicer! ❤️